extensions/cli/cli.go

```diff
@@ -0,0 +1,109 @@
+package cli
+
+import (
+	ranchercli "github.com/rancher/shepherd/clients/ranchercli"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+)
+
+// SwitchContext will display the current context and switch to the default one.
+func SwitchContext(session *session.Session) error {
+	client := &ranchercli.Client{
+		Session: session,
+	}
+
+	logrus.Infof("Listing the current context...")
+	err := client.Create("./rancher", "context", "current")
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Switching to the default context...")
+	err = client.Create("./rancher", "context", "switch")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CreateAndDeleteProjects will create and delete projects in the local cluster.
+func CreateAndDeleteProjects(session *session.Session) error {
+	client := &ranchercli.Client{
+		Session: session,
+	}
+
+	logrus.Infof("Creating a project...")
+	err := client.Create("./rancher", "projects", "create", "--cluster", "local", "auto-cli-project")
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Deleting the project...")
+	err = client.Create("./rancher", "projects", "delete", "auto-cli-project")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CreateAndDeleteNamespaces will create and delete namespaces in the local cluster.
+func CreateAndDeleteNamespaces(session *session.Session) error {
+	client := &ranchercli.Client{
+		Session: session,
+	}
+
+	logrus.Infof("Creating a namespace in default project...")
+	err := client.Create("./rancher", "namespaces", "create", "auto-cli-ns")
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Creating test project...")
+	err = client.Create("./rancher", "projects", "create", "--cluster", "local", "auto-cli-project")
+	if err != nil {
+		return err
+	}
```

Comment:
we talked offline a little about this but I think it would be worth adding a negative test case here too. I.e. this line could be a good candidate, try to create a project in a non-existing cluster and make sure that the client doesn't report a false positive.

Reply:
Implemented an Exists function that should address this.